walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,110 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.*;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+
+    Collection<TimeRange> availableTimeRange = new ArrayList<>();
+    Collection<String> requestAttendees = request.getAttendees();
+    ArrayList<TimeRange> knownEventTimeRange = new ArrayList<>();
+
+    int startOfDay = TimeRange.START_OF_DAY;
+    int endOfDay = TimeRange.END_OF_DAY;
+    long durationOfRequestMeeting = request.getDuration();
+
+    if(durationOfRequestMeeting > TimeRange.WHOLE_DAY.duration()){
+
+        return availableTimeRange;
+
+    } else if (events.size() == 0){
+
+        TimeRange totalTime = TimeRange.fromStartEnd(startOfDay, endOfDay, true);
+        availableTimeRange.add(totalTime);
+        return availableTimeRange;
+
+    }
+
+    
+    for(Event event : events){
+        Set<String> attendees = event.getAttendees();
+
+        //Check if every attendee in each event will be attending the requested meeting
+        int countAttendees = 0;
+        for(String attendee : attendees){
+            if(requestAttendees.contains(attendee)){
+                countAttendees++;
+            }
+        }
+        if(countAttendees == attendees.size()){
+            knownEventTimeRange.add(event.getWhen());
+        }
```

Comment:
You've implemented this correctly so there's no problem with your code.  I just want to point out that Java actually has a method which can do it for you: https://docs.oracle.com/javase/8/docs/api/java/util/Collection.html#containsAll-java.util.Collection-

You could probably write something like this

```
if(requestAttendees.containsAll(event.getAttendees()) {
  knownEventTimeRange.add(event.getWhen());
}
```

No need to change it, just FYI.